tests/unit/rankers/test_algorithms.py

```diff
@@ -145,11 +145,11 @@ def test_spread_publishers_cannot_spread(self):
         recs[2].publisher = 'donnie.com'
         recs[3].publisher = 'donnie.com'
         recs[4].publisher = 'thedude.com'
-        recs[5].publisher = 'thedude.com'
-        recs[6].publisher = 'abides.com'
+        recs[5].publisher = 'abides.com'
+        recs[6].publisher = 'thedude.com'
```

Comment:
This seems like a bug in the old `spread_publishers` code. Moving `thedude.com` down one position increases the distance to the rec with the same domain above it, without bringing `abides.com` within distance 3.